apps/server/src/modules/deletion/repo/deletion-request.repo.ts

```diff
@@ -47,12 +49,24 @@ export class DeletionRequestRepo {
 		return mapped;
 	}
 
-	async countPendingDeletionRequests(): Promise<number> {
-		const scope = new DeletionRequestScope().byStatusPending();
+	async findAllFailedItems(limit: number, olderThan: Date, newerThan: Date): Promise<DeletionRequest[]> {
+		if (olderThan < newerThan) {
+			throw new Error('olderThan must be greater than newerThan');
+		}
```

Comment:
Shouldn't this check better be moved to the service?

Reply:
Was on purpose because the method in service is very basic. 
I consider some date params is not business logic but just common sense which can be normal for a query.
find all records between one week ago and 6 months ago - makes sense
find all records newer than one week ago and older than 6 months ago makes no sense.